File: RealLifeUi/Html/Componente/Login/DivLoginBase.cs
using NetZ.Web.Html;
using NetZ.Web.Html.Componente;
using NetZ.Web.Server.Arquivo.Css;

namespace RealLifeUi.Html.Componente.Login
{
    internal abstract class DivLoginBase : ComponenteHtml
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private Div _divConteudo;

        protected Div divConteudo
        {
            get
            {
                if (_divConteudo != null)
                {
                    return _divConteudo;
                }

                _divConteudo = new Div();

                return _divConteudo;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = this.GetType().Name;
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divConteudo.setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setHeight(45, "vh"));
            this.addCss(css.setPaddingLeft(10, "vw"));
            this.addCss(css.setPaddingRight(10, "vw"));

            this.addCss(css.setAlignItems("center"));
            this.addCss(css.setDisplay("flex"));
            this.addCss(css.setJustifyContent("center"));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: RealLiteUi.TypeScript/html/componente/login/DivLoginBase.ts
/// <reference path="../../../../Web.TypeScript/html/componente/ComponenteHtml.ts"/>
/// <reference path="../../../../web.typescript/typedefinition/md5.d.ts" />
/// <reference path="../../../../Web.TypeScript/Utils.ts"/>

module RealLifeUi
{
    // #region Importações

    import ComponenteHtml = Web.ComponenteHtml;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export abstract class DivLoginBase extends ComponenteHtml
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos
        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        protected encriptar(strSenha: string): string
        {
            if (Utils.getBooStrVazia(strSenha))
            {
                return null;
            }

            var strSenhaEncriptadaResultado = "GTA_Real_Life__senha";

            strSenhaEncriptadaResultado = strSenhaEncriptadaResultado.replace("_senha", strSenha);

            strSenhaEncriptadaResultado = md5(strSenhaEncriptadaResultado);

            return strSenhaEncriptadaResultado;
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}